dds/DCPS/ThreadPool.h

```diff
@@ -10,11 +10,25 @@
 
 #include "dcps_export.h"
 
+#include "ConditionVariable.h"
 #include "PoolAllocator.h"
 
-#include <ace/Barrier.h>
 #include <ace/Thread.h>
 
+#if defined ACE_HAS_THREADS
```

Comment:
Couldn't this be simplified?  We can assume the generic HAS_THREADS is there (or else nothing would work) and ignore the STHREADS for Solaris.  So OPENDDS_NO_THREAD_JOIN is only true if ACE has pthreads and lacks pthread join.

Reply:
I tried a simpler version before and got it wrong so I just duplicated what I was seeing in OS_NS_Thread.cpp for when it was returning `NOTSUP`. I'll try just checking for pthreads and lacks join.